ui/src/app/views/quick-tour-modal/component.tsx

```diff
@@ -0,0 +1,222 @@
+                    </div>}
+                  </div></>;
+              })}
+          </div>
+          <div style={{width: '100%', paddingTop: '5%'}}>
+             <div className='divider'></div>
+          </div>
+          <div className='panel'>
+            <div className='panel-left'>
+              <div className='panel-title'>{panels[this.state.selected].title}</div>
+              <div className='panel-contents'>
+                <div className='panel-text'>{panels[this.state.selected].content}</div>
+              </div>
+            </div>
+            <div className='panel-right'>
+              <img src={panels[this.state.selected].image} className='panel-image'/>
+                {(this.state.selected !== 0) &&
+                <div style={{position: 'absolute', right: '5%',
+                    bottom: '5%', height: '1rem', width: '1rem'}}>
+                    <div className='resize-icon' style={{position: 'absolute', zIndex: 2}}
+                         onClick={() => this.toggleImage()}>
+                        <img src={this.expandIcon}/>
+                    </div>
+                </div>}
+
+            </div>
+          </div>
+          <div className='controls'>
+            <div className='left'>
+              {this.state.selected !== 0 &&
+              <button type='button' className='btn btn-close' id='previous'
+                      onClick={() => this.previous()}>Previous</button>}
+            </div>
+            <div className='right'>
+              {this.state.selected !== (this.state.numPanels - 1) &&
+              <button type='button' className='btn btn-close' id='close'
+                      onClick={() => this.close()}>Close</button>}
+              <button type='button' className='btn btn-primary' id='next'
+                      onClick={() => this.next()}>{this.lastButtonText()}</button>
+            </div>
+          </div>
+        </div>}
+        {this.props.learning && this.state.fullImage && <div className='main full-image'>
+          <div className='full-image-wrapper'>
+            <img src={panels[this.state.selected].image} style={{height: '100%', width: '100%'}}/>
+              <div className='resize-icon' onClick={() => this.toggleImage()}
+                   style={{position: 'absolute', right: '5%', bottom: '5%'}}>
+                  <img src={this.expandIcon}/>
+              </div>
+          </div>
+        </div>}
+      </>;
+  }
+}
+
+@Component({
+  selector: 'app-quick-tour-modal',
+  styleUrls: ['./component.css'],
+  templateUrl: './component.html'
+})
+
+export class QuickTourModalComponent implements OnInit {
+
+  @Input('learning')
+  learning: boolean;
+  @Input('onClose')
+  public onClose: Function;
+
+  constructor() {}
+
+  ngOnInit(): void {
+    ReactDOM.render(React.createElement(QuickTourReact,
+        {learning: this.learning, closeFunction: this.onClose}),
+        document.getElementById('quick-tour'));
+  }
+
+  ngDoCheck(): void {
```

Comment:
Interesting .. this would probably get invoked a bunch. Is this necessary? Wouldn't any state updates necessarily go through `setState()` which would then trigger an update?

Reply:
i think you're right. let me test that.

Reply:
@calbach so it seems like we do need to call the react component through the `ngDoCheck` method, otherwise we initialize it once on `ngOnInit` but then can't actually open it, since the click to access the quick tour is an angular action.   i think lots of things may have to be set up this way as we do half-angular, half-react, but like the angular wrapper, it's a stopgap during the transition process

Reply:
There are two kinds of updates that cause re-renders in react: `state`, which lives and changes inside a component, and `props`, which are passed in from outside. Very often, the state of one component gets passed down as props to another, so the component will see changes to those props over time. This mechanism is basically a systematic way to pass 'state' from angular into the react component as props.

There may be another lifecycle hook that's more appropriate, but the outcome is very much intended: every time the relevant data changes, refresh (e.g. re-render) the react component with updated props.